pkg/services/folder/folderimpl/folder.go

```diff
@@ -261,27 +262,36 @@ func (s *Service) UpdateFolder(ctx context.Context, user *user.SignedInUser, org
 	return nil
 }
 
-func (s *Service) DeleteFolder(ctx context.Context, user *user.SignedInUser, orgID int64, uid string, forceDeleteRules bool) (*models.Folder, error) {
-	dashFolder, err := s.dashboardStore.GetFolderByUID(ctx, orgID, uid)
+func (s *Service) DeleteFolder(ctx context.Context, cmd *folder.DeleteFolderCommand) (*folder.Folder, error) {
+	user, err := appcontext.User(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	guard := guardian.New(ctx, dashFolder.Id, orgID, user)
+	dashFolder, err := s.dashboardStore.GetFolderByUID(ctx, cmd.OrgID, cmd.UID)
+	if err != nil {
+		return nil, err
+	}
+
+	guard := guardian.New(ctx, dashFolder.Id, cmd.OrgID, user)
 	if canSave, err := guard.CanDelete(); err != nil || !canSave {
 		if err != nil {
 			return nil, toFolderError(err)
 		}
 		return nil, dashboards.ErrFolderAccessDenied
 	}
+	if dashFolder == nil {
+		return nil, nil
+	}
```

Comment:
why we need this, given we have already checked in line 272 for errors returned from `GetFolderByUID()` and [GetFolderByUID()](https://github.com/grafana/grafana/blob/0315f6317e2653da203c9ac0f21eceb9aa9e29de/pkg/services/dashboards/database/database.go#L120) only returns `nil` folder in case of errors.
But in any case, I don't think we have to shallow the error if we return `nil` folder.